:gear: add sentry-ruby to gemfile

ShanaLMoore opened the pull request:
I previously set the SENTRY_DSN for each environment but did not notice sentry wasn't part of the Gemfile. Our other projects use sentry-raven which is deprecated, so this commit installs [sentry-ruby](https://github.com/getsentry/sentry-ruby) to track errors in different enviornments.

Follow up for Issue:
- https://github.com/scientist-softserv/ams/issues/76

ref: 
- https://github.com/scientist-softserv/ams/issues/89

Sentry-ruby docs [recommends](https://github.com/getsentry/sentry-ruby#getting-started) installing the following. I added all but delayed_job and resque since we use sidekiq instead. I also excluded "sentry-opentelemetry". At a minimum, sentry-ruby and sentry-rails is required.

```ruby
gem "sentry-rails"
gem "sentry-sidekiq"
gem "sentry-delayed_job"
gem "sentry-resque"
gem "sentry-opentelemetry"
```

Do we need a config file? (tested locally, it seems like we do)

```ruby
# config/initializer/sentry.rb
Sentry.init do |config|
  config.dsn = ENV['SENTRY_DSN']
  config.environment = ENV['SENTRY_ENVIRONMENT']
end

```

I was able to successfully trigger and capture an error from my local environment after doing so: 

![image](https://github.com/WGBH-MLA/ams/assets/10081604/66090480-990a-43bd-ad13-308ed00045c0)

# Notes

FYI: performance monitoring and tracing systems like Sentry, the term ["traces sample rate"](https://docs.sentry.io/platforms/javascript/performance/#configure-the-sample-rate) refers to the proportion or percentage of requests or transactions that are actually captured and sent to the monitoring system for analysis. It's a way to manage the amount of data being collected and transmitted while still gaining insights into the performance of your application.

For example, if you have a traces sample rate of 10%, it means that only 10% of the requests your application receives will have their performance data (traces) collected and sent to the monitoring system like Sentry. The remaining 90% of requests will not be captured, reducing the overall data volume and potential impact on your application's performance and network resources.

A common approach could be to set the traces sample rate in staging to around 30-50% initially, and then monitor the performance impact. You can then adjust the sample rate up or down based on the factors (like: performance impact, budget, traffic volume, debugging needs, and resource usage) and how well your staging environment can handle the additional load.

So in staging I'm setting it to 100% for now. We can monitor and adjust the env variable. In prod I set it to 10% as requested by Rob.

Commit:
:gear: add sentry-ruby to gemfile

I previously set the SENTRY_DSN for each environment but did not notice sentry wasn't part of the Gemfile. Our other projects use sentry-raven which is deprecated, so this commit installs sentry-ruby.

Follow up for Issue:
- https://github.com/scientist-softserv/ams/issues/76

ShanaLMoore commented on Gemfile:
```diff
@@ -123,6 +123,9 @@ gem 'database_cleaner'
 gem 'redlock', '~> 1.0'
 gem 'httparty', '~> 0.21'
 
+# Sentry-raven for error handling
```
```suggestion
# Sentry-ruby for error handling
```

Commit:
Update Gemfile

Commit:
:gear: add sentry-rails

Per the sentry instructions, sentry-rails should be added to the gemfile too.

ref:
- https://scientist-inc.sentry.io/projects/gbh/getting-started/

Commit:
:gear: add sentry config file

Locally I was successful at capturing an error in sentry after including this file.

Commit:
:gear: specs shouldn't report errors to sentry

Commit:
uninstall sentry-opentelemetry

Commit:
:broom: require 'sentry-ruby'

docker build was failing on  NameError: uninitialized constant Sentry

Commit:
Update Gemfile

Commit:
Merge branch 'develop' into i76-follow-up

orangewolf commented on Gemfile:
```diff
@@ -123,6 +123,12 @@ gem 'database_cleaner'
+gem "sentry-ruby"
+gem "sentry-rails"
+gem "sentry-sidekiq"
+gem "sentry-resque"
```
why resque? we are only using sidekiq here I beleive

ShanaLMoore replied:
Thanks for confirming. I got its description mixed up with sentry-ruby and didn't realize it was another [background worker](https://github.com/resque/resque) at the time. I'll remove it. 


Resque is a Redis-backed Ruby library for creating background jobs, placing them on multiple queues, and processing them later.

orangewolf commented on config/initializers/sentry.rb:
```diff
@@ -0,0 +1,17 @@
+  # Set traces_sample_rate to 1.0 to capture 100%
+  # of transactions for performance monitoring.
+  # We recommend adjusting this value in production.
+  config.traces_sample_rate = 1.0
```
this should be 0.1 for prod please

ShanaLMoore replied:
hard coded or as an environment variable? I'll make it an env unless otherwise stated, so we can easily change it live if need be.

Commit:
:broom: remove resque and add env variable SENTRY_TRACES_SAMPLE_RATE

In this commit, we remove resque because it's another background worker; we're using sidekiq so there isn't a need for it. We also set an env variable SENTRY_TRACES_SAMPLE_RATE for a sentry configuration.

Commit:
set higher trace sample rate value for staging environments

Per Sentry 'While testing, set tracesSampleRate to 1.0 as that ensures that every transaction will be sent to Sentry.'

ShanaLMoore commented:
FYI: performance monitoring and tracing systems like Sentry, the term "traces sample rate" refers to the proportion or percentage of requests or transactions that are actually captured and sent to the monitoring system for analysis. It's a way to manage the amount of data being collected and transmitted while still gaining insights into the performance of your application.

For example, if you have a traces sample rate of 10%, it means that only 10% of the requests your application receives will have their performance data (traces) collected and sent to the monitoring system like Sentry. The remaining 90% of requests will not be captured, reducing the overall data volume and potential impact on your application's performance and network resources.

A common approach could be to set the traces sample rate in staging to around 30-50% initially, and then monitor the performance impact. You can then adjust the sample rate up or down based on the factors (like: performance impact, budget, traffic volume, debugging needs, and resource usage) and how well your staging environment can handle the additional load.

So in staging I'm setting it to 100% for now. We can monitor and adjust the env variable. In prod I set it to 10% as requested by Rob.
